csw-server/src/main/java/org/fao/geonet/component/csw/GetRecords.java

```diff
@@ -571,7 +571,7 @@ private String checkTypenames(Element query) throws MissingParameterValueEx, Inv
         }
         // missing typeNames element
         else {
-            return cswPrefix + ":Record";
+            throw new MissingParameterValueEx("typeNames","Attribute 'typeNames' is missing. Add typeNames=\"gmd:MD_Metadata\" to Query Element.");
```

Comment:
Why ? csw:Record is the default value no ? ... even if element is mandatory
![image](https://cloud.githubusercontent.com/assets/1701393/7269205/f070459a-e8d1-11e4-9943-215b3ee4dc22.png)

And we are not only supporting gmd:MD_Metadata so the exception should contains all supported typenames ?